ocaml/xapi/xapi_pool.ml

```diff
@@ -52,8 +52,11 @@ let get_pool ~rpc ~session_id =
   | [] ->
       let err_msg = "Remote host does not belong to a pool." in
       raise Api_errors.(Server_error (internal_error, [err_msg]))
-  | pool :: _ ->
+  | [pool] ->
       pool
+  | _ ->
```

Comment:
```
let internal_error fmt =
  Printf.ksprintf
    (fun str ->
      error "%s" str ;
      raise Api_errors.(Server_error (internal_error, [str]))
    )
    fmt
```

Add this to the file and use it. Report the actual number of pools found and probably `__FUNCTION__` as well